### PR TITLE
doc: add docs of github action usage

### PR DESCRIPTION
**Pull request by frostming**
Signed-off-by: Frost Ming <me@frostming.com>

## Before submitting:

- [ ] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
      guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [ ] Does the code follow BentoML's code style, `pre-commit run -a` script has passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [ ] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
      those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

### CODE REVIEW

File: docs/source/tutorial.rst [hunk 1]
```diff
@@ -650,6 +650,43 @@ specific for ML, the BentoML team has also created Yatai and bentoctl:
 Learn more about different deployment options with BentoML from the
 :doc:`concepts/deploy` page.
 
+Use GitHub Action to Boost Your Workflow
+----------------------------------------
+
+BentoML provides a GitHub Action to help you automate the process of building bento and deploying to the cloud.
```
Comment (Sherlock113) [inline docs/source/tutorial.rst:656]:
```suggestion
BentoML provides a GitHub Action to help you automate the process of building Bentos and deploying them to the cloud.
```

File: docs/source/tutorial.rst [hunk 2]
```diff
@@ -650,6 +650,43 @@ specific for ML, the BentoML team has also created Yatai and bentoctl:
...
+Here an example of the workflow file:
```
Comment (Sherlock113) [inline docs/source/tutorial.rst:657]:
```suggestion
To create a GitHub Actions workflow, you need to first define a workflow file as below:
```

File: docs/source/tutorial.rst [hunk 3]
```diff
@@ -650,6 +650,43 @@ specific for ML, the BentoML team has also created Yatai and bentoctl:
...
+            cache: 'pip'
+
+          - name: Build and Deploy
+            uses: bentoml/deploy-bento-action@main
+              with:
+                cloud_api_token: ${{ secrets.CLOUD_API_TOKEN }}
+                cloud_endpoint: ${{ secrets.CLOUD_ENDPOINT }}
+                deployment_name: test-iris
+
+You can get more details about this file in :ref:`Deploying Your Bento <bentocloud/how-tos/deploy-bentos:Deploying Your Bento>`.
```
Comment (Sherlock113) [inline docs/source/tutorial.rst:685]:
```suggestion
You can get more details about this file in :ref:`Deploying Your Bento <bentocloud/how-tos/deploy-bentos:Deploying Your Bento>` and `the GitHub Actions documentation <https://docs.github.com/en/actions/quickstart>`_.
```

File: docs/source/tutorial.rst [hunk 4]
```diff
@@ -650,6 +650,43 @@ specific for ML, the BentoML team has also created Yatai and bentoctl:
 Learn more about different deployment options with BentoML from the
 :doc:`concepts/deploy` page.
 
+Use GitHub Action to Boost Your Workflow
```
Comment (Sherlock113) [inline docs/source/tutorial.rst:653]:
```suggestion
Using GitHub Actions to Boost Your Workflow
```

**Review by Sherlock113 — COMMENTED** [review]
The added content can be added to this tutorial for now. As it should be applicable to all of our similar tutorials, I prefer to have it in a separate doc. I will restructure this and create a new doc after the reorg of the Overview and Quickstarts sections.

File: docs/source/tutorial.rst [hunk 5]
```diff
@@ -650,6 +650,43 @@ specific for ML, the BentoML team has also created Yatai and bentoctl:
...
+----------------------------------------
+
+BentoML provides a GitHub Action to help you automate the process of building bento and deploying to the cloud.
+Here an example of the workflow file:
+
+.. code-block:: yaml
```
Comment (aarnphm) [inline docs/source/tutorial.rst:659]: I think we should make this a yaml file, and the link it here to rest for us to easier to maintain. As working in yaml file is better than working with yaml within rst

File: docs/source/tutorial.rst [hunk 6]
```diff
@@ -650,6 +650,20 @@ specific for ML, the BentoML team has also created Yatai and bentoctl:
...
+Using GitHub Actions to Boost Your Workflow
+-------------------------------------------
+
+BentoML provides a GitHub Action to help you automate the process of building Bentos and deploying them to the cloud.
+To create a GitHub Actions workflow, you need to first define a workflow file as below:
+
+.. literalinclude:: ./data/workflow.yaml
+   :language: yaml
+
+You can get more details about this file in :ref:`Deploying Your Bento <bentocloud/how-tos/deploy-bentos:Deploying Your Bento>` and `the GitHub Actions documentation <https://docs.github.com/en/actions/quickstart>`_.
```
Comment (aarnphm) [inline docs/source/tutorial.rst:662]:
```suggestion
The action [`bentoml/deploy-bento-action`](https://github.com/bentoml/deploy-bento-action) requires `cloud_api_token` and `cloud_endpoint`, which you can set the [repository secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository) to make sure the job has correct privilege.

You can get more details about this file in :ref:`Deploying Your Bento <bentocloud/how-tos/deploy-bentos:Deploying Your Bento>` and `the GitHub Actions documentation <https://docs.github.com/en/actions/quickstart>`_.
```

**Review by aarnphm — COMMENTED** [review]
one more comment and it is GTG

**Review by aarnphm — APPROVED** [review]

**Pull request merged**